lib/onc_certification_g10_test_kit/tasks/generate_matrix.rb

```diff
@@ -188,6 +188,18 @@ def generate_test_procedure_worksheet # rubocop:disable Metrics/CyclomaticComple
         end
       end
 
+      # returns an array of options that apply to this test or group
+      def applicable_options(runnable)
+        runnable_and_parents = [runnable].tap do |parents|
+          while runnable.respond_to?(:parent) &&
+                runnable.respond_to?(:suite_option_requirements) &&
+                (runnable = runnable.parent)
```

Comment:
```suggestion
                (runnable == runnable.parent)
```
I'm assuming you're intending to do a comparison rather than an assignment here

Reply:
Nope, this is just bad code that i wrote quickly.  This is what allowed the while loop to traverse upwards.  I updated the code to be a bit simpler.

I'm not sure if this has to be this complicated though, I just wasn't sure how much options got pushed down to children in the test tree... so i just had it gather all relevant options for itself and all of its ancestors and then eventually do a 'uniq' to throw out any duplicates.